### PR TITLE
BreakpointWidget: Fix delete deleting both MBP and BP at address

### PR DESCRIPTION
**Pull request by sepalani**
This PR fixes the delete action which deletes everything matching the address of the selected item (i.e. both memory and regular breakpoints).

Ready to be reviewed & merged.

### CODE REVIEW

File: Source/Core/DolphinQt/Debugger/BreakpointWidget.cpp [hunk 1]
```diff
@@ -240,15 +252,24 @@ void BreakpointWidget::Update()
...
+  const auto address = item->data(ADDRESS_ROLE).toUInt();
+  const bool is_memcheck = item->data(IS_MEMCHECK_ROLE).toBool();
 
-  PowerPC::breakpoints.Remove(address);
-  Settings::Instance().blockSignals(true);
-  PowerPC::memchecks.Remove(address);
-  Settings::Instance().blockSignals(false);
+  if (is_memcheck)
+  {
+    Settings::Instance().blockSignals(true);
```
Comment (leoetlino) [inline Source/Core/DolphinQt/Debugger/BreakpointWidget.cpp:265]: I realise this was already written like this, but using a QSignalBlocker would be cleaner here. Idea for a follow-up PR :)

File: Source/Core/DolphinQt/Debugger/BreakpointWidget.cpp [hunk 2]
```diff
@@ -317,48 +338,41 @@ void BreakpointWidget::OnContextMenu()
   }
 
   const auto& selected_item = selected_items.constFirst();
-  const auto bp_address = static_cast<u32>(selected_item->data(Qt::UserRole).toUInt());
-
-  auto is_memory_breakpoint = false;
+  const auto bp_address = static_cast<u32>(selected_item->data(ADDRESS_ROLE).toUInt());
+  auto is_memory_breakpoint = selected_item->data(IS_MEMCHECK_ROLE).toBool();
```
Comment (leoetlino) [inline Source/Core/DolphinQt/Debugger/BreakpointWidget.cpp:342]: Why is this randomly not const?
```suggestion
  const auto is_memory_breakpoint = selected_item->data(IS_MEMCHECK_ROLE).toBool();
```

File: Source/Core/DolphinQt/Debugger/BreakpointWidget.cpp [hunk 3]
```diff
@@ -317,48 +338,41 @@ void BreakpointWidget::OnContextMenu()
...
+        std::find_if(inst_breakpoints.begin(), inst_breakpoints.end(),
+                     [bp_address](const auto& bp) { return bp.address == bp_address; });
+    if (bp_iter == inst_breakpoints.end())
+      return;
+
     menu->addAction(bp_iter->is_enabled ? tr("Disable") : tr("Enable"), [this, &bp_address]() {
       PowerPC::breakpoints.ToggleBreakPoint(bp_address);
       Update();
     });
+    menu->addAction(tr("Go to"), [this, bp_address]() { emit SelectedBreakpoint(bp_address); });
```
Comment (leoetlino) [inline Source/Core/DolphinQt/Debugger/BreakpointWidget.cpp:359]:
```suggestion
    menu->addAction(tr("Go to"), [this, bp_address] { emit SelectedBreakpoint(bp_address); });
```

**Review by leoetlino — CHANGES_REQUESTED** [review]

**Comment by sepalani** [comment]
@leoetlino 
Done.

**Review by leoetlino — APPROVED** [review]

**Pull request merged**